main.py

```diff
@@ -21,6 +31,36 @@ def callback(request):
                 jsonify(message="Currently, the system do not accept a GET request"),
                 405,
             )
-    except IndexError:
+    except:
         return jsonify(message="Something went wrong!"), 400
+
     return "Success"
+
+
+def retry_failed_webhook(transaction_log_service):
+    failed_ivr_logs = transaction_log_service.get_failed_ivr_transaction_log()
+
+    for log in failed_ivr_logs:
+        log.attempts += 1
+        db.session.add(log)
+        db.session.commit()
+
+        payload = json.loads(log.payload)
+        payload["log_created_on"] = log.created_on
+        processed = process_form_data(payload)
+
+        if processed is not True:
+            continue
+
+        log.processed = True
+        db.session.add(log)
+        db.session.commit()
```

Comment:
We can reduce the number of database query per execution. Thoughts?
```suggestion
    for log in failed_ivr_logs:
        payload = json.loads(log.payload)
        payload["log_created_on"] = log.created_on
        log.processed = process_form_data(payload)
        log.attempts += 1
        db.session.add(log)
        db.session.commit()
```

Reply:
In this way, if the logic is not updating the status?

Reply:
Yea, the `log.processed` should be either True or False. And if you see`process_form_data(payload)`, it returns a boolean value. So the status will be based on the return type. Do you think this doesn't work?